purchase_sale_inter_company/models/stock_move.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2023 ForgeFlow
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    def _update_extra_data_in_move(self, move):
+        for rec in self:
+            if hasattr(rec, "_cal_move_weight"):  # from delivery module
+                rec._cal_move_weight()
```

Comment:
`_cal_move_weight()` it's a compute function, are you sure that you really need to call that manually?

Also, there is no need to iterate over `self`, since `_cal_move_weight()` already does that.
I would do something like:
```suggestion
    def _update_extra_data_in_move(self, move):
        if hasattr(self, "_cal_move_weight"):  # from delivery module
            self._cal_move_weight()
```

Also, do you want to update the data of `move` or `self` in that method?